rcljava/src/test/java/org/ros2/rcljava/SpinTest.java

```diff
@@ -0,0 +1,335 @@
+/* Copyright 2020 Open Source Robotics Foundation, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ros2.rcljava;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.ros2.rcljava.RCLJava;
+import org.ros2.rcljava.concurrent.Callback;
+import org.ros2.rcljava.executors.Executor;
+import org.ros2.rcljava.executors.SingleThreadedExecutor;
+import org.ros2.rcljava.node.ComposableNode;
+import org.ros2.rcljava.node.Node;
+import org.ros2.rcljava.timer.WallTimer;
+
+public class SpinTest {
+  public static class TimerCallback implements Callback {
+    private int counter;
+    private final long timeToSleepMs;
+
+    TimerCallback(long sleepTimeMs) {
+      timeToSleepMs = sleepTimeMs;
+    }
+
+    public void call() {
+      this.counter++;
+      try {
+        // The timer is configured to wake up after 100 milliseconds, so wait 200
+        // which should be enough to make it trigger.
```

Comment:
This comment seems disjoint from the implementation.

Reply:
Yeah, good call.  Copy-n-pasted comment from elsewhere, it doesn't belong here.  Removed in 1316feb.